File: Framework.Repositories/Interface/Core/IPageSectionItemRepository.py
from abc import ABC, abstractmethod
from typing import List, Optional, Any
from datetime import datetime
class IPageSectionItemRepository(ABC):
    """
    Python interface equivalent to the C# IPageSectionItemRepository.
    Defines the contract for repository operations.
    """
    @abstractmethod
    def get_page_section_items_by_page_section_id(self, page_section_id): int) -> IEnumerable<PageSectionItem>: # Basic return type mapping
        ...

File: Framework.Repositories/Interface/Core/IPageSectionItemRepository.cs
using Data;
using Framework.Data.Models.Conteudo;
using System.Collections.Generic;
using System.Linq;

namespace Framework.Repositories.Interface.Core
{
    public interface IPageSectionItemRepository : IRepository<PageSectionItem>
    {
        IEnumerable<PageSectionItem> GetPageSectionItemsByPageSectionId(int pageSectionId);
    }

    public class PageSectionItemRepository : Repository<PageSectionItem>, IPageSectionItemRepository
    {
        public PageSectionItemRepository(ApplicationDbContext context) : base(context)
        {
        }

        public IEnumerable<PageSectionItem> GetPageSectionItemsByPageSectionId(int pageSectionId)
        {
            return _context.PageSectionItem
                .Where(psi => psi.PageSectionId == pageSectionId && !psi.Excluido)
                //.Include(psi => psi.StyleProperties)
                .OrderBy(psi => psi.Ordem)
                .ToList();
        }
    }

}
